### PR TITLE
Add inject and extract for TextMap

### PR DESCRIPTION
**Pull request by bensigelman**
Manual tests happened; automated tests did not.

@bcronin I didn't see a place to add automated tests for the `common/` code, so I decided to wait for instructions about how to do so before adding such things. Or should I just add them on the JRE side and leave it at that?

Manual testing did reveal one important bug (I'll point it out).

### CODE REVIEW

File: common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java [hunk 1]
```diff
@@ -532,7 +547,6 @@ protected void addTracerTag(String key, String value) {
       SpanRecord record = new SpanRecord();
       record.setSpan_name(this.operationName);
       record.setOldest_micros(this.startTimestampMicros);
-      record.setSpan_guid(generateGUID());
```
Comment (bensigelman) [inline common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java]: (the bug)

File: common/src/main/java/com/lightstep/tracer/shared/Propagator.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import io.opentracing.propagation.TextMapWriter;
+
+public interface Propagator {
+  boolean matchesInjectCarrier(Object carrier);
+  boolean matchesExtractCarrier(Object carrier);
+  void inject(SpanContext spanContext, Object carrier);
+  SpanContext extract(Object carrier);
+
+  // The three supported Propagators.
+  static final Propagator TEXT_MAP = new Propagator() {
```
Comment (bensigelman) [inline common/src/main/java/com/lightstep/tracer/shared/Propagator.java]: this inline class is the only interesting thing in this PR (aside from the bug)

**Comment by bensigelman** [comment]
I figured out how to install the particular minor version of the android build tools that we depend on, removed some unused imports, and am going to merge this.

**Comment by bensigelman** [comment]
(and (try to) push a new minor version of our artifacts)

**Comment by bcronin** [comment]
The `common/` directory doesn't build directly to anything, so that's correct that adding tests to the JRE build is easiest.

**Pull request merged**